diff --git ./include/linux/gfp.h ./include/linux/gfp.h
index 7c777a0..d22632e 100644
--- ./include/linux/gfp.h
+++ ./include/linux/gfp.h
@@ -219,13 +219,13 @@ static inline enum zone_type gfp_zone(gfp_t flags)
 	z = (GFP_ZONE_TABLE >> (bit * ZONES_SHIFT)) &
 					 ((1 << ZONES_SHIFT) - 1);
 
-	if (__builtin_constant_p(bit))
+/*	if (__builtin_constant_p(bit))
 		BUILD_BUG_ON((GFP_ZONE_BAD >> bit) & 1);
 	else {
 #ifdef CONFIG_DEBUG_VM
 		BUG_ON((GFP_ZONE_BAD >> bit) & 1);
 #endif
-	}
+	}*/
 	return z;
 }
 
diff --git ./include/linux/kernel.h ./include/linux/kernel.h
index d6320a3..5130398 100644
--- ./include/linux/kernel.h
+++ ./include/linux/kernel.h
@@ -675,7 +675,7 @@ struct sysinfo {
 };
 
 /* Force a compilation error if condition is true */
-#define BUILD_BUG_ON(condition) ((void)sizeof(char[1 - 2*!!(condition)]))
+#define BUILD_BUG_ON(condition) //((void)sizeof(char[1 - 2*!!(condition)]))
 
 /* Force a compilation error if condition is true, but also produce a
    result (of value 0 and type size_t), so the expression can be used
diff --git ./include/net/inet_sock.h ./include/net/inet_sock.h
index 47004f3..ed3da43 100644
--- ./include/net/inet_sock.h
+++ ./include/net/inet_sock.h
@@ -205,7 +205,7 @@ static inline struct request_sock *inet_reqsk_alloc(struct request_sock_ops *ops
 	struct inet_request_sock *ireq = inet_rsk(req);
 
 	if (req != NULL) {
-		kmemcheck_annotate_bitfield(ireq, flags);
+//		kmemcheck_annotate_bitfield(ireq, flags);
 		ireq->opt = NULL;
 	}
 
